lib/puppet/ssl/host.rb

```diff
@@ -192,14 +194,23 @@ def generate_certificate_request(options = {})
     true
   end
 
+  def http_client
+    @http_client ||= Puppet::Rest::Client.new(Puppet::Rest::Routes.ca)
+  end
+
```

Comment:
For now, I've tied the lifetime of the client to the host making the requests. This seemed more straightforward than attempting to thread it through from the `application` entry point. If we decide it makes more sense to create one client in some large scope and reuse it throughout (one argument I've heard against that is the potential to bleed different security states between requests), we can remove this.

Reply:
Looks like this might not be an option, apparently Puppetserver uses the host class too, which means it's attempting to require the HTTPClient gem, though I'm pretty sure we don't use it for cert downloading. I'm looking into options.